frontend/ui/widget/inputdialog.lua

```diff
@@ -723,6 +723,76 @@ function InputDialog:_addScrollButtons(nav_bar)
+            table.insert(row, {
+                text = "Go",
+                callback = function()
+                    -- calculate current line number
+                    local curr_line_num = 1
+                    for i = 1, #self._input_widget.charlist do
+                        if self._input_widget.charlist[i] == "\n" then
+                            if i >= self._input_widget.charpos then break end
+                            curr_line_num = curr_line_num + 1
+                        end
+                    end
+                    local last_line_num = select(2, string.gsub(self:getInputText(), "\n", "")) + 1
+                    input_dialog = InputDialog:new{
+                        title = _("Enter line number"),
+                        input_hint = T("%1 (1 - %2)", curr_line_num, last_line_num),
+                        input_type = "number",
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(input_dialog)
+                                        if not self.keyboard_hidden then
+                                            self:onShowKeyboard()
+                                        end
+                                    end,
+                                },
+                                {
+                                    text = _("Go to line"),
+                                    is_enter_default = true,
+                                    callback = function()
+                                        local new_line_num = tonumber(input_dialog:getInputText())
+                                        if new_line_num and new_line_num >= 1 and new_line_num <= last_line_num then
+                                            -- calculate charpos for the beginning of the target line
+                                            local new_char_pos = 1
+                                            if new_line_num > 1 then
+                                                local j = 1
+                                                for i = 1, #self._input_widget.charlist do
+                                                    if self._input_widget.charlist[i] == "\n" then
+                                                        j = j + 1
+                                                        if j == new_line_num then
+                                                            new_char_pos = i + 1
+                                                            break
+                                                        end
+                                                    end
+                                                end
+                                            end
+                                            -- reinit with the target line on the top, cursor at the beginning
```

Comment:
I don't mind as this would be rarely used, but feels like a re-init is not needed.
You could add this stuff as methods of InputText:
`InputText:scrollToLineNum()`
and ideally, the line num should be in the middle, so a bit above it and a bit below it are shown ?

The other stuff could may be also be put in InputText:
`InputText:getCurLineNum() return cur_line_num, total_line_nb end`

Reply:
Thank you, I'll look at it.

Reply:
> You could add this stuff as methods of InputText

Done, it is much better.
No init() needed, now Go button is available in the Tweak document settings and everywhere with enabled `add_nav_bar`.

Reply:
> and ideally, the line num should be in the middle, so a bit above it and a bit below it are shown ?

Now, if the target line is out of the screen, when jumping forward the target line is dispalyed last on the screen, when jumping backward - first.
To show the target line in the middle requires some additions to the ScrollTextWidget and TextBoxWidget. Is it worth?

Reply:
I am quite good with the current behaviour.
It does not require to search a cursor in the middle of the screen to trace the target line.

Reply:
> Is it worth?

Probably not :) You're the main user, you decide :)